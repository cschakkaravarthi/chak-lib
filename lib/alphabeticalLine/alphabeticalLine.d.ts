import React from 'react';
import '../../scss/_alphabetical-line.scss';
declare type Props = {
    letter: string;
    className?: string;
    isActive?: string | boolean;
    onClick?: (event: React.MouseEvent<HTMLElement>) => void;
};
export declare const AlphabeticalLine: React.FC<Props>;
export {};
//# sourceMappingURL=alphabeticalLine.d.ts.map