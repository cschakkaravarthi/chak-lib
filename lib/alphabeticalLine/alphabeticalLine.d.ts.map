{"version":3,"file":"alphabeticalLine.d.ts","sourceRoot":"","sources":["../../src/alphabeticalLine/alphabeticalLine.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAG1B,OAAO,oCAAoC,CAAC;AAE5C,aAAK,KAAK,GAAG;IACX,MAAM,EAAE,MAAM,CAAC;IACf,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,QAAQ,CAAC,EAAE,MAAM,GAAG,OAAO,CAAC;IAC5B,OAAO,CAAC,EAAE,CAAC,KAAK,EAAE,KAAK,CAAC,UAAU,CAAC,WAAW,CAAC,KAAK,IAAI,CAAC;CAC1D,CAAC;AAEF,eAAO,MAAM,gBAAgB,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAc5C,CAAC"}