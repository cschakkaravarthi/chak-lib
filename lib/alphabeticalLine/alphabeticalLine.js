"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var Button_1 = __importDefault(require("react-bootstrap/Button"));
require("../../scss/_alphabetical-line.scss");
exports.AlphabeticalLine = function (props) {
    return (react_1["default"].createElement(Button_1["default"], { variant: "link", id: props.letter, value: props.letter, onClick: props.onClick, className: "font-weight-bolder " + props.className + " " + (props.isActive ? 'is-active' : '') }, props.letter));
};
//# sourceMappingURL=alphabeticalLine.js.map