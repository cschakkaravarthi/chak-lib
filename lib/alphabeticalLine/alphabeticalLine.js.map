{"version":3,"file":"alphabeticalLine.js","sourceRoot":"","sources":["../../src/alphabeticalLine/alphabeticalLine.tsx"],"names":[],"mappings":";;;;;AAAA,gDAA0B;AAC1B,kEAA4C;AAE5C,8CAA4C;AAS/B,QAAA,gBAAgB,GAAoB,UAAC,KAAK;IACrD,OAAO,CACL,iCAAC,mBAAM,IACL,OAAO,EAAC,MAAM,EACd,EAAE,EAAE,KAAK,CAAC,MAAM,EAChB,KAAK,EAAE,KAAK,CAAC,MAAM,EACnB,OAAO,EAAE,KAAK,CAAC,OAAO,EACtB,SAAS,EAAE,wBAAsB,KAAK,CAAC,SAAS,UAC9C,KAAK,CAAC,QAAQ,CAAC,CAAC,CAAC,WAAW,CAAC,CAAC,CAAC,EAAE,CACjC,IAED,KAAK,CAAC,MAAM,CACN,CACV,CAAC;AACJ,CAAC,CAAC"}