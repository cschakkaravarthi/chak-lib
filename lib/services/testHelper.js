"use strict";
/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var shallow_1 = __importDefault(require("react-test-renderer/shallow"));
exports.shallowRender = function (toRender) {
    var renderer = shallow_1["default"].createRenderer();
    renderer.render(toRender);
    return renderer.getRenderOutput();
};
//# sourceMappingURL=testHelper.js.map