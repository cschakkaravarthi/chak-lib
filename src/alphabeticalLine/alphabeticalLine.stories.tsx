import React, { useState } from 'react';
import StoriesContainer from '../StoriesContainer';
import { AlphabeticalLine } from './alphabeticalLine';
import Container from 'react-bootstrap/Container';

export const alphabet = [
  'a',
  'b',
  'c',
  'd',
  'e',
  'f',
  'g',
  'h',
  'i',
  'j',
  'k',
  'l',
  'm',
  'n',
  'o',
  'p',
  'q',
  'r',
  's',
  't',
  'u',
  'v',
  'w',
  'x',
  'y',
  'z'
];

export default {
  component: AlphabeticalLine,
  title: 'Alphabetical Line'
};

export const alphabeticalLine = (): JSX.Element => {
  const [isActive, setIsActive] = useState('');

  const handleActiveItem = (letter: string): void => {
    setIsActive(letter);
  };

  return (<StoriesContainer>
    <p>
        This component takes 3 props: letter, onClick and isActive. You can play with them
        to set the active class to the proper element or just to grab the value of the
        clicked element and fire any other event onClick.
    </p>
    <hr/>
    <Container
      className="py-3 alphabetical-line d-flex flex-wrap justify-content-between">
      {alphabet.map((item: string) => (
        <AlphabeticalLine
          letter={item.toUpperCase()} key={item} onClick={() => handleActiveItem(item)} isActive={item === isActive} />))}
    </Container>
  </StoriesContainer>);
};
