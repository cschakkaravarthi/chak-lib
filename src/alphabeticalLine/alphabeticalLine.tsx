import React from 'react';
import Button from 'react-bootstrap/Button';

import '../../scss/_alphabetical-line.scss';

type Props = {
  letter: string;
  className?: string;
  isActive?: string | boolean;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
};

export const AlphabeticalLine: React.FC<Props> = (props): JSX.Element => {
  return (
    <Button
      variant="link"
      id={props.letter}
      value={props.letter}
      onClick={props.onClick}
      className={`font-weight-bolder ${props.className} ${
        props.isActive ? 'is-active' : ''
      }`}
    >
      {props.letter}
    </Button>
  );
};
