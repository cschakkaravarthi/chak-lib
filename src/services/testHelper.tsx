/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.

import ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRender = (toRender: any): Record<string, any> => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};
