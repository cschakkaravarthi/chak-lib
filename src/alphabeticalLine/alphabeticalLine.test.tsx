import React from 'react';

import { shallowRender } from '../services/testHelper';
import { AlphabeticalLine } from './alphabeticalLine';

describe('Resources', () => {
  it('should render correctly', () => {
    const tree = shallowRender(
      <AlphabeticalLine
        letter="A"
        onClick={e => console.log((e.target as HTMLInputElement).value)} />

    );
    expect(tree).toMatchSnapshot();
  });
});
