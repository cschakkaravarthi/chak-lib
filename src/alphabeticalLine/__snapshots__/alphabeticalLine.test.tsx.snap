// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources should render correctly 1`] = `
<ForwardRef
  active={false}
  className="font-weight-bolder undefined "
  disabled={false}
  id="A"
  onClick={[Function]}
  type="button"
  value="A"
  variant="link"
>
  A
</ForwardRef>
`;
